mypy/test/testerrorstream.py

```diff
@@ -0,0 +1,82 @@
+"""Tests for mypy incremental error output."""
+from typing import List, Callable, Optional
+
+import os
+
+from mypy import defaults, build
+from mypy.test.config import test_temp_dir
+from mypy.myunit import AssertionFailure
+from mypy.test.helpers import assert_string_arrays_equal
+from mypy.test.data import DataDrivenTestCase, DataSuite
+from mypy.build import BuildSource
+from mypy.errors import CompileError
+from mypy.options import Options
+from mypy.plugin import Plugin, ChainedPlugin, DefaultPlugin, FunctionContext
+from mypy.nodes import CallExpr, StrExpr
+from mypy.types import Type
+
+
+class ErrorStreamSuite(DataSuite):
+    files = ['errorstream.test']
+
+    def run_case(self, testcase: DataDrivenTestCase) -> None:
+        test_error_stream(testcase)
+
+
+def test_error_stream(testcase: DataDrivenTestCase) -> None:
+    """Perform a single error streaming test case.
+
+    The argument contains the description of the test case.
+    """
+    options = Options()
+    options.show_traceback = True
+
+    logged_messages = []  # type: List[str]
+    real_messages = []  # type: List[str]
+
+    def flush_errors(msgs: List[str], serious: bool, is_real: bool=True) -> None:
+        if msgs:
+            logged_messages.append('==== Errors flushed ====')
+            logged_messages.extend(msgs)
+        if is_real:
+            real_messages.extend(msgs)
+
+    plugin = ChainedPlugin(options, [LoggingPlugin(options, flush_errors), DefaultPlugin(options)])
+
+    sources = [BuildSource('main', '__main__', '\n'.join(testcase.input))]
+    try:
+        res = build.build(sources=sources,
+                          options=options,
+                          alt_lib_path=test_temp_dir,
+                          flush_errors=flush_errors,
+                          plugin=plugin)
```

Comment:
This is the only place where we pass a plugin to `build()`. I wish there was another way to do this... (It's pretty clever though. :-)

Reply:
Another approach would be arrange to load a plugin through the regular mechanisms---that requires the plugin communicating with the test via global state, though.

Another another approach is to ditch the plugin because the possible failure tests for (errors being streamed out properly but not until all typing is done) seems pretty marginal anyways.

Reply:
I'm okay with ditching the plugin.